pkgs/applications/window-managers/herbstluftwm/default.nix

```diff
@@ -39,8 +40,29 @@ stdenv.mkDerivation rec {
     libXrandr
   ];
 
+  doCheck = true;
+
+  checkInputs = [
+    (python3.withPackages (ps: with ps; [ ewmh pytest xlib ]))
+    xdotool
+    xorg.xorgserver
+    xorg.xsetroot
+    xterm
+  ];
+
+  patches = [
+    ./test-path-environment.patch
+  ];
+
   postPatch = ''
     patchShebangs doc/gendoc.py
+    substituteInPlace tests/conftest.py --replace "/usr/bin/env bash" ${runtimeShell}
+    substituteInPlace tests/test_herbstluftwm.py --replace "/usr/bin/env bash" ${runtimeShell}
+    substituteInPlace share/autostart --replace /etc/xdg/herbstluftwm $out/etc/xdg/herbstluftwm
+  '';
+
+  checkPhase = ''
+    python3 -m pytest ../tests
```

Comment:
Please use the ``python.interpreter``.

Reply:
I would need to use `(python3.withPackages (ps: with ps; [ ewmh pytest xlib ])).interpreter` to make it works (or wrap it with a `let … in`). Is that what you want?